lib/structuraid_core/engineering/optimization/rebar_constant_long.rb

```diff
@@ -0,0 +1,126 @@
+require 'interactor'
+
+module StructuraidCore
+  module Engineering
+    module Optimization
+      # Runs optimization of the reinforcement, it asumes that the rebar length is constant, so if any shosen rebar has the same length as any other, the target is just to minimize the total mass by changing the amount of rebars
+      class RebarConstantLong
+        include Interactor
+
+        OPTIONAL_REBAR_NUMBERS = [3, 4, 5, 6, 7].freeze
+        MINIMUM_REBAR_AMOUNT = 2
+        UNSUCCESSFUL_RESULT_MESSAGE = 'required area exceeds maximum spacing'.freeze
+        SUCCESSFUL_RESULT_MESSAGE = 'done'.freeze
+
+        # @param required_reinforcement_area [Float] The reinforcement area required by the design
+        # @param maximum_rebar_spacing [Float] The maximum spacing between rebars
+        # @param space_to_cover [Float] The space to cover with the reinforcement, the rebars will be placed with a spacing which must be less than maximum_rebar_spacing
+        def initialize(required_reinforcement_area, maximum_rebar_spacing, space_to_cover)
+          @required_reinforcement_area = required_reinforcement_area
+          @maximum_rebar_spacing = maximum_rebar_spacing
```

Comment:
should we consider a minimum rebar spacing as well? Just thinking of possible scenarios beyond footings where we could have a minimum spacing.

Also following the same idea, I think the logic inside this class should consider the maximum or minimum spacing optional parameters, which would make it more reusable